test/camelCaseSyntax.spec.js

```diff
@@ -0,0 +1,226 @@
+        expect(subject).toEqual('foo')
+      );
+      expect(clonedExpect('foo').toFoo).notToBeAFunction();
+    });
+  });
+
+  describe('in a child', () => {
+    it('should succeed', () => {
+      expect.child()(456).toBeGreaterThan(123);
+    });
+
+    it('should fail', () => {
+      expect(() => expect.child()(123).toBeGreaterThan(456)).toThrow(
+        'expected 123 to be greater than 456'
+      );
+    });
+
+    it('should support an assertion subsequently added to the parent', () => {
+      const parentExpect = expect.clone();
+      const childExpect = parentExpect.child();
+      parentExpect.addAssertion('<string> to foo', (expect, subject) =>
+        expect(subject).toEqual('foo')
+      );
+      childExpect('foo').toFoo();
+    });
+  });
+
+  describe('with an assertion that is defined for the base of the given subject type', () => {
+    it('should succeed', () => {
+      function foo() {}
+      foo.bar = 123;
+      expect(foo).toHaveProperty('bar', 123);
+    });
+
+    it('should fail', () => {
+      function foo() {}
+      expect(() => expect(foo).toHaveProperty('bar', 123)).toThrow(
+        "expected function foo() {} to have property 'bar', 123"
+      );
+    });
+  });
+
+  describe('with an assertion that is not defined for the given subject type', () => {
+    it('should not expose a function', () => {
+      expect(expect(123).toContain).toBeUndefined();
+    });
+  });
+
+  describe('with a middle rocket assertion', () => {
+    it('should succeed', () => {
+      expect([1, 2, 3]).whenPassedAsParametersTo(Math.max).toEqual(3);
+    });
+
+    it('should fail', () => {
+      expect(() =>
+        expect([1, 2, 3]).whenPassedAsParametersTo(Math.max).toEqual(2)
+      ).toThrow('expected 3 to equal 2');
+    });
+  });
+
+  describe('with the expect.it equivalent', () => {
+    it('should succeed', () => {
+      expect({ foo: 123 }).toSatisfy({ foo: expect.toBeANumber() });
+    });
+
+    describe('and extra chaining with .and...', () => {
+      it('should succeed', () => {
+        expect({ foo: 123 }).toSatisfy({
+          foo: expect.toBeANumber().andToBeGreaterThan(100),
+        });
+      });
+
+      it('should fail', () => {
+        expect(() =>
+          expect({ foo: 123 }).toSatisfy({
+            foo: expect.toBeANumber().andToBeGreaterThan(200),
+          })
+        ).toThrow(
+          'expected { foo: 123 } to satisfy\n' +
+            '{\n' +
+            "  foo: expect.it('to be a number')\n" +
+            "               .and('to be greater than', 200)\n" +
+            '}\n' +
+            '\n' +
+            '{\n' +
+            '  foo: 123 // ✓ should be a number and\n' +
+            '           // ⨯ should be greater than 200\n' +
+            '}'
+        );
+      });
+    });
+
+    describe('and extra chaining with .or...', () => {
+      it('should succeed', () => {
+        expect({ foo: 123 }).toSatisfy({
+          foo: expect.toBeAString().orToBeGreaterThan(100),
```

Comment:
Im not too keen of having different spelling of the assertions like this - I think that would get confusing when it came to documentation and the like.

Thinking aloud were ok chaining through a .or. we could potentially generate the object with all these assertions only once.

Reply:
That's fair, we have a bunch of wiggle room wrt. the syntax here. Again, I'm not too keen on mixing dot notation and camel case, but something like `expect.toBeAString().or().toBeGreaterThan(100)` could be arranged?

Reply:
That is far better to my eyes at least!